docs/tutorials/add-route.md

```diff
@@ -0,0 +1,163 @@
+```php
+App::post('/v1/storage/buckets/:bucketId/files')
+    ->alias('/v1/storage/files', ['bucketId' => 'default'])
+```
+
+### 2. Description
+Used as an abstract description of the route. 
+```php
+App::post('/v1/storage/buckets/:bucketId/files')
+->desc('Create File')
+```
+
+### 3. Groups
+Groups array is used to group one or more routes with one or more hooks functionality.
+```php
+App::post('/v1/storage/buckets/:bucketId/files')
+->groups(['api'])
+```
+In the above example groups() is used to define the current route as part of the routes that shares a common init middleware hook. 
+```php
+App::init()
+->groups(['api'])
+->action(
+  some code.....
+);
+```
+
+
+### 4. The Labels Mechanism
+Labels are very strait forward and easy to use and understand, but in the same time are very robust.
+Labels are passed from the controllers route and used to pick up key-value pairs to be handled in a centralized place
+along the road.
+Labels can be used to pass a pattern in order to be replaced in the other end.
+Appwrite uses different labels to achieve different things, for example:
+
+#### Scope
+* scope - Defines the route permissions scope.
+
+```php
+App::post('/v1/storage/buckets/:bucketId/files')
+    ->label('scope', 'files.write')
+```
+
+#### Audit
+* audits.event - Identify the log in human-readable text.
+* audits.userId - Signals the extraction of $userId in places that it's not available natively.
+* audits.resource - Signals the extraction part of the resource.
+
+
+```php
+App::post('/v1/account/create')
+->label('audits.event', 'account.create')
+->label('audits.resource', 'user/{response.$id}')
+->label('audits.userId', '{response.$id}')
+```
+
+#### Sdk
+* sdk.auth - Array of authentication types is passed in order to impose different authentication methods in different situations.
+* sdk.namespace - Refers to the route namespace.
+* sdk.method - Refers to the sdk method that needs to called.
+* sdk.description - Description of the route,using markdown format.
+* sdk.sdk.response.code - Refers to the route http response status code expected.
+* sdk.auth.response.model - Refers the route http response expected.
+
+```php
+App::post('/v1/account/jwt')
+->label('sdk.auth', [APP_AUTH_TYPE_SESSION])
+->label('sdk.namespace', 'account')
+->label('sdk.method', 'createJWT')
+->label('sdk.description', '/docs/references/account/create-jwt.md')
+->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+->label('sdk.response.model', Response::MODEL_JWT)
+```
+
+#### Cache
+* cache - When set to true, signal the use of file cache. 
+* cache.resource - Identifies the cached resource.
+
+```php
+App::get('/v1/storage/buckets/:bucketId/files/:fileId/preview')
+->label('cache', true)
+->label('cache.resource', 'file/{request.fileId}')
+```
+
+#### Abuse
+* abuse-key - Specifies routes unique abuse key.
```

Comment:
Would be great to list a few of the supported vars, or briefly explain how they work.